File: 프로그래머스/2/12949. 행렬의 곱셈/행렬의 곱셈.java
class Solution {
    public int[][] solution(int[][] arr1, int[][] arr2) {
        int[][] answer = new int[arr1.length][arr2[0].length];
        
        for(int m = 0; m < arr1.length; m++) {
            for(int n = 0; n < arr2[0].length; n++){
                for(int k = 0; k < arr1[0].length; k++){
                    answer[m][n] += (arr1[m][k] * arr2[k][n]);
                }
            }
        }
        
        return answer;
    }
}

File: 프로그래머스/2/12949. 행렬의 곱셈/행렬의 곱셈.py
def solution(arr1, arr2):    
    r1, c1 = len(arr1), len(arr1[0])    
    r2, c2 = len(arr2), len(arr2[0])
    
    answer = [[0]*c2 for _ in range(r1)]
    
    for x in range(r1):
        for z in range(c2):
            for y in range(c1):
                answer[x][z] += arr1[x][y] * arr2[y][z] 


    return answer